lib/openstax/aws/secrets.rb

```diff
@@ -27,17 +33,77 @@ def create(specification:, substitutions: {})
       # Ship 'em
       if !dry_run
         built_secrets.each do |built_secret|
-          client.put_parameter(built_secret)
+          client.put_parameter(built_secret.merge(overwrite: true))
+        end
+      end
+    end
+
+    def update(specifications: nil, substitutions: nil)
+      existing_secrets = data!
+
+      built_secrets = build_secrets(specifications: specifications, substitutions: substitutions)
+
+      changed_secrets = built_secrets.each_with_object([]) do |built_secret, array|
```

Comment:
`array` is the name of the array passed into `each_with_object([])` using the naming pattern like in https://blog.arkency.com/inject-vs-each-with-object/ 's use of a variable named `hash`.  This iterator is building up an array, and that array is `array`.  The array is what is assigned to `changed_secrets`, so I don't think it should be `built_secrets`.

Reply:
array i guess is ok.  just wanted to to describe as much info as it can.  array of what.  

perhaps it's an "accumulated_changed_secrets".    

i mention this because i've seen a lot of each_with_object app code that thru poor naming was easy to introduce a bug because one might have to look up which of these 2 params is what or forget to.  

no biggie. array is fine if that's what you want.